### PR TITLE
add group filter for user api

### PR DESCRIPTION
**Pull request by RolandMa1986**
Signed-off-by: Roland.Ma <rolandma@yunify.com>

**What type of PR is this?**
/kind api-change

**What this PR does / why we need it**:
Frontend need to filter user by user's group
/assign @wansir

### CODE REVIEW

File: pkg/models/resources/v1alpha3/user/users.go [hunk 1]
```diff
@@ -107,6 +108,10 @@ func (d *usersGetter) filter(object runtime.Object, filter query.Filter) bool {
 	switch filter.Field {
 	case iamv1alpha2.FieldEmail:
 		return user.Spec.Email == string(filter.Value)
+	case iamv1alpha2.InGroup:
```
Comment (wansir) [inline pkg/models/resources/v1alpha3/user/users.go:111]: What happened if `user.spec.group` has duplicate content.

Comment (RolandMa1986) [reply pkg/models/resources/v1alpha3/user/users.go:111]: There's no duplication here, The group name is unique in the cluster.

**Pull request merged**